Sources/TuistKit/Services/InitService.swift

```diff
@@ -92,23 +99,50 @@ class InitService {
 
         let directories = try templatesDirectoryLocator.templateDirectories(at: path)
         if let templateName = templateName {
-            guard
-                let templateDirectory = directories.first(where: { $0.basename == templateName })
-            else { throw InitServiceError.templateNotFound(templateName) }
-            let template = try templateLoader.loadTemplate(at: templateDirectory)
-            let parsedAttributes = try parseAttributes(
-                name: name,
-                platform: platform,
-                requiredTemplateOptions: requiredTemplateOptions,
-                optionalTemplateOptions: optionalTemplateOptions,
-                template: template
-            )
+            var template: Template?
+            var parsedAttributes: [String: String]
+            if templateName.isGitURL {
```

Comment:
there's a bunch of logic here (testing for `isGitURL`, `parseRepoBranch`, etc.) and it'd be nice to get some coverage here in the unit tests